src/integrations/blocks/structured-data-blocks.php

```diff
@@ -173,6 +173,75 @@ public function optimize_faq_images( $attributes, $content ) {
 		return $this->optimize_images( $attributes['questions'], 'answer', $content );
 	}
 
+	/**
+	 * Transforms the durations into a translated string containing the count, and either singular or plural unit.
+	 * For example (in en-US): If 'days' is 1, it returns "1 day". If 'days' is 2, it returns "2 days".
+	 * If a number value is 0, we don't output the string.
+	 *
+	 * @param number $days Number of days.
+	 * @param number $hours Number of hours.
+	 * @param number $minutes Number of minutes.
+	 * @return array Array of pluralized durations.
+	 */
+	private function transform_duration_to_string( $days, $hours, $minutes ) {
+		$strings = [];
+		if ( \intval( $days ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of day/days. */
+				\_n( '%d day', '%d days', $days, 'wordpress-seo' ),
+				$days
+			);
+		}
+		if ( \intval( $hours ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of hour/hours. */
+				\_n( '%d hour', '%d hours', $hours, 'wordpress-seo' ),
+				$hours
+			);
+		}
+		if ( \intval( $minutes ) !== 0 ) {
+			$strings[] = \sprintf(
+			/* translators: %d expands to the number of minute/minutes. */
+				\_n( '%d minute', '%d minutes', $minutes, 'wordpress-seo' ),
+				$minutes
+			);
+		}
+		return $strings;
+	}
+
+	/**
+	 * Formats the durations into a translated string.
+	 *
+	 * @param array $attributes The attributes.
+	 * @return string The formatted duration.
+	 */
+	private function build_duration_string( $attributes ) {
+		$days            = ( isset( $attributes['days'] ) ) ? ( $attributes['days'] ) : ( 0 );
+		$hours           = ( isset( $attributes['hours'] ) ) ? ( $attributes['hours'] ) : ( 0 );
+		$minutes         = ( isset( $attributes['minutes'] ) ) ? ( $attributes['minutes'] ) : ( 0 );
+		$elements        = $this->transform_duration_to_string( $days, $hours, $minutes );
+		$elements_length = count( $elements );
+
+		if ( $elements_length === 1 ) {
+			return $elements[0];
+		}
+		if ( $elements_length === 2 ) {
+			return \sprintf(
+			/* translators: %s expands to a unit of time (e.g. 1 day). */
+				\__( '%1$s and %2$s', 'wordpress-seo' ),
+				...$elements
+			);
+		}
+		if ( $elements_length === 3 ) {
+			return \sprintf(
+			/* translators: %s expands to a unit of time (e.g. 1 day). */
+				\__( '%1$s, %2$s and %3$s', 'wordpress-seo' ),
+				...$elements
+			);
+		}
+		return '';
+	}
+
```

Comment:
The logic to build and format the duration text is replicated from `packages/js/src/structured-data-blocks/how-to/components/HowTo.js`